src/screen.c

```diff
@@ -21,28 +21,30 @@ void screen_init()
 void screen_clear()
 {
   for (u16 x = 0; x < VGA_WIDTH; x++)
-  {
     for (u16 y = 0; y < VGA_HEIGHT; y++)
-    {
-      display_buffer[screen_position_to_index(x, y)] = '\0' | (7 << 8);
-    }
-  }
+      display_buffer[screen_position_to_index(x,y)]  = '\0' | (0 << 8);
 }
 
 void screen_set_cursor_position(i8 x, i8 y)
 {
-  u16 index = screen_position_to_index(x, y + 1);
+  u16 index = (u16)(y * VGA_WIDTH + x);
 
   out8(0x3D4, 0x0F);
   out8(0x3D5, (u8)(index & 0xFF));
   out8(0x3D4, 0x0E);
   out8(0x3D5, (u8)((index >> 8) & 0xFF));
 }
 
-void screen_draw_char_at(u8 x, u8 y, char c)
+void screen_draw_char_at(u8 x, u8 y,u8 forecolour, u8 backcolour,u8 lightmode, char c)
```

Comment:
Lightmode is not needed. Also, please stick to the naming convention and spacing. Also, use bools in the future.